code/modules/xenomorph/silo.dm

```diff
@@ -153,3 +153,11 @@
 /obj/structure/xeno/silo/proc/update_minimap_icon()
 	SSminimaps.remove_marker(src)
 	SSminimaps.add_marker(src, MINIMAP_FLAG_XENO, image('icons/UI_icons/map_blips.dmi', null, "silo[warning ? "_warn" : "_passive"]", HIGH_FLOAT_LAYER))
+
+/// Transfers the spawned minion to the silo's hivenumber.
+/obj/structure/xeno/silo/proc/on_spawn(list/newly_spawned_things)
+	for(var/spawned_thing AS in newly_spawned_things)
+		if(!isxeno(spawned_thing))
+			CRASH("Xeno silo somehow tried to spawn a non xeno (tried to spawn [spawned_thing])")
+		var/mob/living/carbon/xenomorph/spawned_minion = spawned_thing
+		spawned_minion.transfer_to_hive(hivenumber)
```

Comment:
AS means the type check is skipped, I dunno why you added a crash it A) breaks if 1 is non a minion and all after will not get transfered since it ends the proc, and B) this istype shouldnt be needed because we shouldnt account for every error ever, it should be reasonable to assume the silo only spawns xeno types

Reply:
old spawner code used CRASH so I kept it around (and added it to silo too). fixed